kork-secrets-gcp/src/main/java/com/netflix/spinnaker/kork/secrets/engines/GoogleCredentials.java

```
@@ -0,0 +1,60 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.kork.secrets.engines;
+
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.HttpBackOffUnsuccessfulResponseHandler;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.util.ExponentialBackOff;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.concurrent.TimeUnit;
+
+public class GoogleCredentials {
+
+  private static final long CONNECT_TIMEOUT = TimeUnit.MINUTES.toMillis(2);
+  private static final long READ_TIMEOUT = TimeUnit.MINUTES.toMillis(2);
+
+  public static HttpTransport buildHttpTransport() {
+    try {
+      return GoogleNetHttpTransport.newTrustedTransport();
+    } catch (GeneralSecurityException | IOException e) {
+      throw new RuntimeException("Failed to build trusted transport", e);
+    }
+  }
+
+  public static HttpRequestInitializer retryRequestInitializer() {
+    return request -> {
+      request.setConnectTimeout((int) CONNECT_TIMEOUT);
+      request.setReadTimeout((int) READ_TIMEOUT);
+      request.setUnsuccessfulResponseHandler(
+          new HttpBackOffUnsuccessfulResponseHandler(new ExponentialBackOff()));
+    };
+  }
+
+  public static HttpRequestInitializer setHttpTimeout(
+      final HttpRequestInitializer requestInitializer) {
+    return request -> {
+      requestInitializer.initialize(request);
+      request.setConnectTimeout((int) CONNECT_TIMEOUT);
+      request.setReadTimeout((int) READ_TIMEOUT);
+      request.setUnsuccessfulResponseHandler(
+          new HttpBackOffUnsuccessfulResponseHandler(new ExponentialBackOff()));
```

Are you sure that we want to retry with exponential backoff on any non-200 response?  What are the backoff parameters created by `ExponentialBackoff`?  I suspect that if a file is missing (or we don't have permission) it will be a better experience to fail fast rather than retry exponentially for a while then fail.

---

It's not immediately obvious but default behavior of `HttpBackOffUnsuccessfulResponseHandler` is to retry only on 5xx errors. I'll set it explicitly for clarity.